fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/Select.java

```diff
@@ -145,6 +161,109 @@ public String toString() {
+     * @return the query string starting XELECT instead of SELECT
+     */
+    public String toDebugString() {
+
+        try {
+            // Generate a pretty-printed string using the renderer so we
+            // get an accurate version of the string.
+            StringStatementRenderer renderer = new StringStatementRenderer(TRANSLATOR, null, true);
+            return this.render(renderer);
+        } catch (Exception x) {
+            // If we can't render, it's very likely this debug is being used already
+            // in a catch clause from an earlier exception. So rather than propagating
+            // another exception, we just make note of the error and allow the earlier
+            // catch deal with reporting the issue
+            return "Failed to render statement: " + x.getMessage();
+        }
+    }
+
+    /**
+     * Render the components of the statement
+     * @param <T>
+     * @param renderer
+     * @return
+     */
+    public <T> T render(StatementRenderer<T> renderer) {
+        return renderer.select(distinct, selectList, fromClause, whereClause, groupByClause, havingClause, orderByClause, paginationClause);
+    }
+
+    /**
+     * Add an inner join to the from clause for this select statement.
+     * @param tableName
+     * @param alias
+     * @param joinOnPredicate
+     */
+    public void addInnerJoin(String tableName, Alias alias, ExpNode joinOnPredicate) {
+        fromClause.addInnerJoin(tableName, alias, joinOnPredicate);
+    }
+
+    /**
+     * Add a left outer join to the from clause for this select statement.
+     * @param tableName
+     * @param alias
+     * @param joinOnPredicate
+     */
+    public void addLeftOuterJoin(String tableName, Alias alias, ExpNode joinOnPredicate) {
+        fromClause.addLeftOuterJoin(tableName, alias, joinOnPredicate);
+    }
+
+    /**
+     * @param ob
+     */
+    public void setOrderByClause(OrderByClause ob) {
+        this.orderByClause = ob;
+    }
+
+    /**
+     * Getter for the whereClause
+     * @return the whereClause value, which can be null if not yet set
+     */
+    public WhereClause getWhereClause() {
+        return this.whereClause;
+    }
+
+    /**
+     * Visit the components of this query
+     * @param <T>
+     * @param visitor
+     * @return
+     */
+    public <T> T visit(ExpNodeVisitor<T> visitor) {
+        T result;
+        if (whereClause != null) {
+            result = whereClause.visit(visitor);
+        } else {
+            result = null;
+        }
+        return result;
+    }
+
+    /**
+     * Add a pagination clause (offset/limit) to the query.
+     * @param offset
+     * @param rowsPerPage
+     */
+    public void addPagination(int offset, int rowsPerPage) {
+        this.paginationClause = new PaginationClause(offset, rowsPerPage);
+    }
+
+    /**
+     * @return the ORDER BY clause object for this select statement. Can be null.
+     */
+    public OrderByClause getOrderByClause() {
```

Comment:
For consistency, might be good to have getters and setters for all of the Clause types.

Reply:
Will be refactoring the Select statement in #2321 to make it an ExpNode which will change how this is done.